### PR TITLE
Docs, code cleanup, and misc. fixes for smallvec

### PR DESCRIPTION
**Pull request by mbrubeck**
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-smallvec/40)

### CODE REVIEW

File: lib.rs [hunk 1]
```diff
@@ -130,13 +150,45 @@ impl<A: Array> Drop for SmallVecData<A> {
     }
 }
 
-
+/// A Vec-like that can store a small number of elements inline.
```
Comment (Ms2ger) [inline lib.rs]:
```
`Vec`-like
```

File: lib.rs [hunk 2]
```diff
@@ -245,6 +310,9 @@ impl<A: Array> SmallVec<A> {
         }
     }
 
+    /// Reserve capacity for at least `additional` more elements to be inserted.
+    ///
+    /// May reserve more space to avoid frequent reallocations.
```
Comment (Ms2ger) [inline lib.rs]: Maybe make a note about overflow behaviour. In particular, it is not safe to assume that you can actually fit `additional` new elements in it!

File: lib.rs [hunk 3]
```diff
@@ -350,9 +350,24 @@ impl<A: Array> SmallVec<A> {
...
     /// Shrink the capacity of the vector as much as possible.
+    ///
+    /// When possible, this will move data from an external heap buffer to the vector's inline
+    /// storage.
     pub fn shrink_to_fit(&mut self) {
         let len = self.len;
-        if self.spilled() && self.capacity() > len {
+        if len <= self.inline_size() {
+            unsafe {
+                let (ptr, capacity) = match &self.data {
```
Comment (Ms2ger) [inline lib.rs]: Why borrow `self.data`?

**Comment by mbrubeck** [comment]
@bors-servo r=Ms2ger

* Addressed review comments
* Added some whitespace cleanup
* Bumped patch version

**Pull request merged**